core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequestTests.java

```diff
@@ -137,4 +171,72 @@ private static CreateIndexRequest createTestItem() throws IOException {
 
         return request;
     }
+
+    public static Settings randomSettings() {
+        Settings.Builder builder = Settings.builder();
+
+        if (randomBoolean()) {
+            int numberOfShards = randomIntBetween(1, 10);
+            builder.put(SETTING_NUMBER_OF_SHARDS, numberOfShards);
+        }
+
+        if (randomBoolean()) {
+            int numberOfReplicas = randomIntBetween(1, 10);
+            builder.put(SETTING_NUMBER_OF_REPLICAS, numberOfReplicas);
+        }
+
+        return builder.build();
+    }
+
+    public static XContentBuilder randomMapping(String type) throws IOException {
+        XContentBuilder builder = XContentFactory.jsonBuilder();
+        builder.startObject().startObject(type);
+
+        randomMappingFields(builder, true);
+
+        builder.endObject().endObject();
+        return builder;
+    }
+
+    private static void randomMappingFields(XContentBuilder builder, boolean allowObjectField) throws IOException {
+        builder.startObject("properties");
+
+        int fieldsNo = randomIntBetween(0, 5);
+        for (int i = 0; i < fieldsNo; i++) {
+            builder.startObject(randomAlphaOfLength(5));
+
+            if (allowObjectField && randomBoolean()) {
+                randomMappingFields(builder, randomBoolean());
```

Comment:
sorry, I see only now that you were already limiting the number of nested objects. My concern was to not end up with too many levels of nested objects. I think passing in false was good here, we do not need complicated mappings in this test.